src/modules/onboarding/signup/index.tsx

```diff
@@ -0,0 +1,113 @@
+import {
+  Stack,
+  Button,
+  TextField,
+  Checkbox,
+  FormControlLabel,
+  InputAdornment,
+  IconButton,
+  Typography,
+} from '@mui/material';
+import { Paper } from '../../../shared-components/Paper/Paper';
+import { styled, Theme } from '@mui/material/styles';
+import { Regular1 } from '../../../shared-components/Regular1';
+import GoogleIcon from '@mui/icons-material/Google';
+import React, { useState } from 'react';
+import Visibility from '@mui/icons-material/Visibility';
+import VisibilityOff from '@mui/icons-material/VisibilityOff';
+import Link from '@mui/material/Link';
+import ArrowBack from '@mui/icons-material/ArrowBack';
+import { Title2 } from '../../../shared-components/Title2';
+import '../signup/style.css';
+
+const StyledDiv = styled(`div`)(({ theme }: { theme: Theme }) => ({
+  display: 'flex',
+  flexDirection: 'column',
+  justifyContent: 'space-between',
+  minHeight: `calc(100vh - ${theme.spacing(3)})`,
+}));
+
+export const SignUp = () => {
+  const [showPassword, setShowPassword] = useState(false);
+
+  const handleClickShowPassword = () => {
+    setShowPassword(!showPassword);
+  };
+
+  const handleMouseDownPassword = (
+    event: React.MouseEvent<HTMLButtonElement>,
+  ) => {
+    event.preventDefault();
+  };
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+  };
+  return (
+    <Paper>
+      <StyledDiv>
+        <Stack direction='row' alignItems='center' spacing={15} sx={{ mb: 2 }}>
+          <IconButton
```

Comment:
Here you need to redirect the user to whatever page he/she were before. To do that, use `useNavigate()` hook from 'react-router-dom'. There's an example in the lesson files in the RecipeDetail component (e.g. Lesson39).